bindgen/templates/ObjectTemplate.cs

```diff
@@ -17,15 +17,31 @@
     {%- endmatch -%}
     {%- endfor %} {
     {%- for meth in obj.methods() %}
+    {%- if !meth.is_async() %}
     {%- call cs::docstring(meth, 4) %}
     {%- call cs::method_throws_annotation(meth.throws_type()) %}
     {%  call cs::return_type(meth) %} {{ meth.name()|fn_name }}({% call cs::arg_list_decl(meth) %});
+    {%- endif %}
     {%- endfor %}
 }
 
 {%- call cs::docstring(obj, 0) %}
-{{ config.access_modifier() }} class {{ impl_name }}: FFIObject, {{ interface_name }} {
-    public {{ impl_name }}(IntPtr pointer) : base(pointer) {}
+{%- if (is_error) %}
+{{ config.access_modifier() }} class {{ impl_name }} : UniffiException, {{ interface_name }}, IDisposable {
+{%- else %}
+{{ config.access_modifier() }} class {{ impl_name }} : {{ interface_name }}, IDisposable {
+{%- endif %}
```

Comment:
Nit: could simplify this to something like
```suggestion
{{ config.access_modifier() }} class {{ impl_name }} : {{ interface_name }}, IDisposable {% if (is_error) %}, UniffiException {% endif %} {
```